cpp/src/arrow/engine/substrait/serde_test.cc

```diff
@@ -216,6 +216,18 @@ void CheckRoundTripResult(const std::shared_ptr<Table> expected_table,
   compute::AssertTablesEqualIgnoringOrder(merged_expected, output_table);
 }
 
+int CountProjectNodeOptionsInDeclarations(const compute::Declaration& input) {
+  static int counter = 0;
```

Comment:
Why static?

Reply:
Ah, I see your point, I updated the logic and also add it as a validation where emit is used.